File: C++/maxProfit.cpp
class Solution {
public:
    int maxProfit(vector<int> &prices) {
        if (prices.size() < 2) return 0;
        int min = INT_MAX;
        int maxP = 0;
        for (size_t i = 0; i < prices.size(); i++) {
            if (prices[i] < min) {
                min = prices[i];
            }
            if (prices[i] - min > maxP) {
                maxP = prices[i] - min;
            }
        }
        
        return maxP;
    }
};

File: Java/maxProfit.java
public class Solution {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE;
        int maxp = 0;

        if (prices.length < 2) return 0; 

        for (int i = 0; i < prices.length; i++) {
            min = Math.min(prices[i], min);
            maxp = Math.max(prices[i] - min, maxp);
        }

        return maxp;
    }
}
